### DISCUSSION TITLE
How to add description to array items?

### DISCUSSION THREAD

**Discussion by risabhds** [microsoft/typespec#4248] [Json Schema] [answered]
I want to generate the following json schema:
```json
{
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "$id": "Demo.json",
    "type": "object",
    "properties": {
        "example": {
            "type": "array",
            "items": {
                "type": "string",
                "description": "some description of string"
            }
        }
    },
    "required": [
        "example"
    ]
}
```
Notice the `description` is with `type: string` and not `type: array`.

But unable to get this using typespec. The description is added on `type: array`.

[Playground link](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvanNvbi1zY2hlbWEiOwoKdXNpbmcgVHlwZVNwZWMuSnNvblPFHTsKCkDELMYOCm5hbWVzcGFjZSDGEXM7CgoKbW9kZWwgRGVtbyB7CiAgLyoqIHNvbWUgZGVzY3JpcHRpb24gb2Ygc3RyxGUqLwogIGV4YW1wbGU6IEFycmF5PMYbPjsKfQ%3D%3D&e=%40typespec%2Fjson-schema&options=%7B%22options%22%3A%7B%22%40typespec%2Fjson-schema%22%3A%7B%22file-type%22%3A%22json%22%7D%7D%7D)

How to add description on the nested items `type: string`?

**Comment by timotheeguerin** [comment, marked as answer]
It is not possible as it is, if you do need that the solution is to create an array model with the description 

```
model Demo {
  example: ArrayOfString;
}

/** some description of string */
model ArrayOfString is string[];
```
[Playground](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvanNvbi1zY2hlbWEiOwoKdXNpbmcgVHlwZVNwZWMuSnNvblPFHTsKCkDELMYOCm5hbWVzcGFjZSDGEXM7CgoKbW9kZWwgRGVtbyB7CiAgZXhhbXBsZTogQXJyYXlPZlN0cmluZzsKfQoKLyoqIHNvbWUgZGVzY3JpcHRpb24gb2Ygc8UjICovx0zNOiBpc8chW107&e=%40typespec%2Fjson-schema&options=%7B%22options%22%3A%7B%22%40typespec%2Fjson-schema%22%3A%7B%22file-type%22%3A%22json%22%7D%7D%7D)

**Reply by risabhds** [reply]
Thanks for the response! I ended up doing the same but was still curious if there was a way to do this without creating a new model.
